src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,56 +1,129 @@
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie = ({ onAdd }: Props) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [movie, setMovie] = useState({
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  });
+
+  const onClickHandler = () => (
+    setMovie({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    })
+  );
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => {
+        event.preventDefault();
+        setCount(count + 1);
+        onAdd(movie);
+        setTitle('');
+        setDescription('');
+        setImgUrl('');
+        setImdbUrl('');
+        setImdbId('');
+      }}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(event) => {
+          setTitle(event);
+        }}
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        value={description}
+        onChange={(event) => {
+          setDescription(event);
+        }}
       />
 
       <TextField
         name="imgUrl"
         label="Image URL"
-        value=""
+        value={imgUrl}
+        required
+        onChange={(event) => {
+          setImgUrl(event);
+        }}
       />
 
       <TextField
         name="imdbUrl"
         label="Imdb URL"
-        value=""
+        value={imdbUrl}
+        required
+        onChange={(event) => {
+          setImdbUrl(event);
+        }}
       />
 
       <TextField
         name="imdbId"
         label="Imdb ID"
-        value=""
+        value={imdbId}
+        required
+        onChange={(event) => {
+          setImdbId(event);
+        }}
       />
 
       <div className="field is-grouped">
         <div className="control">
-          <button
-            type="submit"
-            data-cy="submit-button"
-            className="button is-link"
-          >
-            Add
-          </button>
+          {(title.length === 0
+          || imgUrl.length === 0
+          || imdbUrl.length === 0
+          || imdbId.length === 0)
+            ? (
+              <button
+                type="submit"
+                data-cy="submit-button"
+                className="button is-link"
+                disabled
+              >
```

Comment:
instead of rendering 2 buttons you can set disabled conditionally
```suggestion
              <button
                type="submit"
                data-cy="submit-button"
                className="button is-link"
                disabled={someBooleanValue}
              >
```